File: prime.hs
import Text.Printf

prime :: Int -> Int
prime n = prime' n 2

prime' :: Int -> Int -> Int
prime' n d
  | d > n `div` 2       = 1
  | n `mod` d == 0      = 0
  | otherwise           = prime' n (d + 1)

count_prime :: Int -> Int
count_prime n
  | n < 2       = 0
  | otherwise   = prime n + count_prime (n - 1)

main = printf "number of primes = %d\n" n
  where n = count_prime 100000

File: prime.rs

fn prime(num: i32) -> bool {
    for i in 2..=num/2 {
        if (num % i) == 0 {
            return false
        }
    }
    return true
}

fn main() {
    let mut num = 0;
    for i in 2..=100000 {
        if prime(i) {
            num = num + 1;
        }
    }
    println!("number of primes = {}", num)
}
